example_test.go

```diff
@@ -0,0 +1,322 @@
+
+	})
+	// Output:
+}
+
+func Example_IsTrue() {
+	runExampleTest(func(t testing.TB) {
+		isValid := func() bool {
+			return true
+		}
+		qt.Assert(t, qt.IsTrue(1 == 1))
+		qt.Assert(t, qt.IsTrue(isValid()))
+	})
+	// Output:
+
+}
+
+func Example_IsFalse() {
+	runExampleTest(func(t testing.TB) {
+		isValid := func() bool {
+			return false
+		}
+		qt.Assert(t, qt.IsFalse(1 == 0))
+		qt.Assert(t, qt.IsFalse(isValid()))
+	})
+	// Output:
+
+}
+
+func Example_Not() {
+	runExampleTest(func(t testing.TB) {
+
+		got := []int{1, 2}
+		qt.Assert(t, qt.Not(qt.IsNil(got)))
+
+		answer := 13
+		qt.Assert(t, qt.Not(qt.Equals(answer, 42)))
+	})
+	// Output:
+}
+
+func Example_StringContains() {
+	runExampleTest(func(t testing.TB) {
+		qt.Assert(t, qt.StringContains("hello world", "hello"))
+	})
+	// Output:
+}
+
+func Example_SliceContains() {
+	runExampleTest(func(t testing.TB) {
+		qt.Assert(t, qt.SliceContains([]int{3, 5, 7, 99}, 99))
+		qt.Assert(t, qt.SliceContains([]string{"a", "cd", "e"}, "cd"))
+	})
+	// Output:
+}
+
+func Example_MapContains() {
+	runExampleTest(func(t testing.TB) {
+		qt.Assert(t, qt.MapContains(map[string]int{
+			"hello": 1234,
+		}, 1234))
+	})
+	// Output:
+}
+
+func Example_SliceAny() {
+	runExampleTest(func(t testing.TB) {
+		qt.Assert(t, qt.SliceAny([]int{3, 5, 7, 99}, qt.F2(qt.Equals[int], 7)))
+		qt.Assert(t, qt.SliceAny([][]string{{"a", "b"}, {"c", "d"}}, qt.F2(qt.DeepEquals[[]string], []string{"c", "d"})))
+	})
+	// Output:
+}
+
+func Example_MapAny() {
+	runExampleTest(func(t testing.TB) {
+		qt.Assert(t, qt.MapAny(map[string]int{"x": 2, "y": 3}, qt.F2(qt.Equals[int], 3)))
+	})
+	// Output:
+}
+
+func Example_SliceAll() {
+	runExampleTest(func(t testing.TB) {
+		qt.Assert(t, qt.SliceAll([]int{3, 5, 8}, func(e int) qt.Checker {
+			return qt.Not(qt.Equals(e, 0))
+		}))
+		qt.Assert(t, qt.SliceAll([][]string{{"a", "b"}, {"a", "b"}}, qt.F2(qt.DeepEquals[[]string], []string{"a", "b"})))
+	})
+	// Output:
+}
+
+func Example_MapAll() {
+	runExampleTest(func(t testing.TB) {
+		qt.Assert(t, qt.MapAll(map[string]int{
+			"x": 2,
+			"y": 2,
+		}, qt.F2(qt.Equals[int], 2)))
+	})
+	// Output:
+}
+
+func Example_JSONEquals() {
+	runExampleTest(func(t testing.TB) {
+		data := `[1, 2, 3]`
+		qt.Assert(t, qt.JSONEquals(data, []uint{1, 2, 3}))
+	})
+	// Output:
+}
+
+func Example_ErrorAs() {
+	runExampleTest(func(t testing.TB) {
+		_, err := os.Open("/non-existent-file")
+
+		// Checking for a specific error type
+		qt.Assert(t, qt.ErrorAs(err, new(*os.PathError)))
+		qt.Assert(t, qt.ErrorAs[*os.PathError](err, nil))
+
+		// Checking fields on a specific error type
+		var pathError *os.PathError
+		if qt.Check(t, qt.ErrorAs(err, &pathError)) {
+			qt.Assert(t, qt.Equals(pathError.Path, "/non-existent-file"))
+		}
+	})
+	// Output:
+}
+
+func Example_ErrorIs() {
+	runExampleTest(func(t testing.TB) {
+		_, err := os.Open("/non-existent-file")
+
+		qt.Assert(t, qt.ErrorIs(err, os.ErrNotExist))
+	})
+	// Output:
```

Comment:
can we add a `fmt.Println("PASS")` at the end of `runExampleTest`, just for the sake of being very clear that the test passed? it also avoids the awkwardness of `// Output:` followed by... nothing :)

Reply:
good idea. Done.